pkg/cmd/util/helpers.go

```diff
@@ -149,6 +151,33 @@ func RetrieveApplicationsByName(ctx context.Context, c client.Client, applicatio
 	return applicationMetaList, nil
 }
 
+func RetrieveApplicationStatusByName(ctx context.Context, c client.Client, applicationName string, namespace string) (ApplicationStatusMeta, error) {
+	var applicationStatusMeta ApplicationStatusMeta
+	var appConfig corev1alpha2.ApplicationConfiguration
+	if err := c.Get(ctx, client.ObjectKey{Name: applicationName, Namespace: namespace}, &appConfig); err != nil {
+		return applicationStatusMeta, err
+	}
+
+	component, err := GetComponent(ctx, c, appConfig.Spec.Components[0].ComponentName, namespace)
```

Comment:
We should list component with componentName == applicationName here, instead of the first one of index.

Reply:
In the currently design, there is only one component in the application, so `appConfig.Spec.Components[0].ComponentName` is equal to `applicationName`. But as you said, I should use `applicationName` here instead of `appConfig.Spec.Components[0].ComponentName`. Thanks.